x/pairing/keeper/unresponsive_provider.go

```diff
@@ -92,7 +92,7 @@ func (k Keeper) UnstakeUnresponsiveProviders(ctx sdk.Context, epochsNumToCheckCU
 				continue
 			}
 			// update the CU count for this provider in providerCuCounterForUnreponsivenessMap
-			providerPaymentStorageKeyList, complaintCU, servicedCU, err := k.countCuForUnresponsiveness(ctx, minPaymentBlock, epochsNumToCheckCUForUnresponsiveProvider, epochsNumToCheckCUForComplainers, providerStakeEntry)
+			epochs, complaintCU, servicedCU, err := k.countCuForUnresponsiveness(ctx, minPaymentBlock, epochsNumToCheckCUForUnresponsiveProvider, epochsNumToCheckCUForComplainers, providerStakeEntry)
```

Comment:
shouldnt we just iterate on the providers that have complainer cu instead of iterating over ALL the providers?

Reply:
done. see https://github.com/lavanet/lava/commit/c4f20a3d6e0e9c6034fe38fac073bd36051dd979